### ISSUE TITLE
Envoy as management target via SkyWalking CDS

### ISSUE THREAD

**Issue by Shikugawa** [envoyproxy/envoy#15023]
Apache Skywalking introduced Configuration Discovery Service(CDS) feature. 

https://github.com/apache/skywalking/issues/6114

It works like Envoy xDS but Skywalking as control plane of Skywalking agent. Today, we can use Envoy as Skywalking agent, then this functionality should be applied to Envoy also. It works not like xDS but establishing unary gRPC periodically, like a healthcheck agent. Currently it is WIP in Skywalking C++ SDK, but I will introduce this feature without using SDK's gRPC native client but using Envoy gRPC because of gRPC abstraction shouldn't be ignored. https://github.com/SkyAPM/cpp2sky/pull/61

**Comment by Shikugawa** [comment]
cc @wu-sheng

**Comment by antoniovicente** [comment]
@Shikugawa are you planning on working on this?

**Comment by Shikugawa** [comment]
@antoniovicente Yes I'm planning to implement this in the future after skywalking protocol on Envoy had updated to 8.4.0.

**Comment by Shikugawa** [comment]
@wbpcode @dio @lizan I think that we can provide dynamic configuration into 

- ignore_suffix ... It ignores to trace if path matched here. We can use this feature without declaring explicitly here.
- sample_n_per_3_secs ... Sample N segments per 3 secs.

These are currently supported on Java agent. I don't think first one doesn't have to have because we can use same feature with current Envoy config. We can't achieve second one with current Envoy config. Do you have any idea that first one should be introduced or not on Envoy's Skywalking config? And do you think which  should be better that second one introduced into Skywalking config or 1 level higher class?

**Comment by wu-sheng** [comment]
@Shikugawa Could you share how Envoy would do this?

**Comment by Shikugawa** [comment]
@wu-sheng Route config has tracing property and invoked by matched route. https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-route

**Comment by wu-sheng** [comment]
If your meaning is about routine settings, some with tracing opened, others are not. Then native configurations are really meaningful.
Changing the routing rules due to ignore rules, which are purely tracing's internal issues, is making the Envoy configurations very complex, and lack of maintainability

**Comment by wbpcode** [comment]
I agree with @wu-sheng. For first feature, It would be better to use Envoy's SkyWalking native config. While a similar effect can be simulated using the configuration in the route, it introduces additional complexity.

**Comment by wbpcode** [comment]
For  the second one, I also tend to use the native configuration of SkyWalking to achieve it. This is not a general configuration of tracer, so at least for the time being, we have no need to abstract it to a higher level. A higher level configuration means a deeper intrusion into Envoy.
